### PR TITLE
Choose Visible Sites

### LINKED ISSUE #98: Add Ability To Choose Which Sites Are Visible
Request from SFTT, add ability to show only young trees, adopted trees, etc.

### PR DESCRIPTION
**Pull request by SofieDunt**
## Why

Closes #98.

Allows users to choose which kinds of sites (young, standard, adopted, open) are visible ON DESKTOP, making the experience more appropriate per the user's needs. Extracts map logic from the MapView useEffect to make future testing easier.

## This PR

Pulled out map logic into appropriate files under `mapPageComponents/logic`.
Created a new component `siteLegend` to contain a checkbox that allows the user to choose which kinds of sites are visible.
- Used the slide down and map legend components - the new component appears as a slide-down container on the map containing both the legend and the checkbox 
  - the layout on mobile was more complex/crowded, will wait on adding this feature to mobile

## Screenshots
Checkbox1
Checkbox2
Hidden

**UPDATE**  
Icons with labels
image
Responsive styling

old -> responisve1

 

## Verification Steps

 
Navigated the map on desktop and mobile, making sure everything worked as expected.

### CODE REVIEW

File: src/components/mapPageComponents/logic/event.ts [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+        event.feature.getProperty('lat'),
+        event.feature.getProperty('lng'),
+      ),
+      map,
+      view,
+    );
+  });
+}
+
+// todo complete
```
Comment (rymaju) [inline src/components/mapPageComponents/logic/event.ts:73]: I would prefer if we removed commented out code but if we're going to get to this soon then its _ok_

File: src/components/mapPageComponents/logic/init.ts [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+/**
+ * Creates the private streets layer and sets its initial style, returns the layer.
+ * @param map the map to add the layer to
+ */
+export function initPrivateStreets(map: google.maps.Map): google.maps.Data {
+  const privateStreetsLayer = new google.maps.Data({ map });
+  // Loads the objects into the layer
+  privateStreetsLayer.loadGeoJson(
+    'https://raw.githubusercontent.com/florisdobber/SFTT-map-test/master/private_streets.json',
```
Comment (rymaju) [inline src/components/mapPageComponents/logic/init.ts:29]: Can we move this into the bundle? Or host this on S3 instead?

**Review by rymaju — APPROVED** [review]
Looks awesome! Couple of questions but overall amazing refactoring 💯

**Review by willmt80 — CHANGES_REQUESTED** [review]
Looks good! Left one question and one small suggestion

File: src/components/forms/stewardshipForm/index.tsx [hunk 3]
```diff
@@ -59,7 +59,7 @@ const StewardshipForm: React.FC<StewardshipFormProps> = ({
         <Form.Item name="activityDate" rules={activitiesDateRules}>
           <TreeDatePicker format={'MM/DD/YYYY'} disabledDate={disabledDate} />
         </Form.Item>
-        <ItemLabel>Stewardship Activites</ItemLabel>
+        <ItemLabel>Stewardship Activities</ItemLabel>
```
[the remark this answers is not in the document]
Comment (rymaju) [reply src/components/forms/stewardshipForm/index.tsx:62]: lmao you should have seen how many times I misspelled 'Announcments' in LLB

File: src/components/mapPageComponents/logic/init.ts (hunk 2 above)
Comment (SofieDunt) [reply src/components/mapPageComponents/logic/init.ts:29]: #101

File: src/components/adoptionDirections/index.tsx [hunk 4]
```diff
@@ -9,9 +9,7 @@ import {
 } from '../../assets/content';
 import { MID_GREEN } from '../../utils/colors';
 
-const { Paragraph, Text, Title } = Typography;
```
[the remark this answers is not in the document]
Comment (SofieDunt) [reply src/components/adoptionDirections/index.tsx:12]: There was a CI reference error issue where it couldn't find Typography.

File: src/components/mapPageComponents/constants.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+};
+export const ALL_SITES_VISIBLE: CheckboxValueType[] = [
+  'Young',
+  'Adopted',
+  'Standard',
+  'Open',
+];
+export const SITE_OPTIONS: { label: ReactNode; value: string }[] = [
+  {
+    label: treeSpan(youngTreeIcon, 'Young Trees'),
```
Comment (willmt80) [inline src/components/mapPageComponents/constants.tsx:50]: 💯

**Review by willmt80 — APPROVED** [review]
💯  Looks good!

**Pull request merged**